docs/reference/modules/gateway.asciidoc

```diff
@@ -56,9 +56,22 @@ NOTE: These settings only take effect on a full cluster restart.
 [[dangling-indices]]
 ==== Dangling indices
 
-When a node joins the cluster, any shards stored in its local data
-directory which do not already exist in the cluster will be imported into the
-cluster. This functionality is intended as a best effort to help users who
-lose all master nodes. If a new master node is started which is unaware of
-the other indices in the cluster, adding the old nodes will cause the old
-indices to be imported, instead of being deleted.
+When a node joins the cluster, if it finds any shards stored in its local data
+directory that do not already exist in the cluster, it will consider those
+shards to be "dangling". You can automatically import dangling indices
+into the cluster using the following setting, however this is not safe, and
+you should use the <<dangling-indices-api,Dangling indices API>> instead.
+
+`gateway.auto_import_dangling_indices`::
+
+    deprecated:[7.9.0, This setting will be removed in 8.0. You should use the dedicated dangling indices API instead.]
+    Whether to automatically import dangling indices into the cluster
+    state, provided no indices already exist with the same name. Defaults
+    to `false`.
+
+WARNING: The auto-import functionality was intended as a best effort to help users
+who lose all master nodes. For example, if a new master node were to be
```

Comment:
I'm wondering if we should add that that scenario (losing all master nodes) should be avoided at all cost, as it puts the cluster's metadata and data at risk. Also worth pointing out above (when talking about "this is not safe") that importing dangling indices does not provide any guarantees as to whether the data that's imported truly represents the latest state of the data as it was when the index was still part of the cluster.

Reply:
@ywelsch I've reworked the text to and address your points.